ansible/roles/hv-network/tasks/main.yml

```
@@ -4,6 +4,7 @@
 - name: Clean lab interfaces
   shell: |
     /root/clean-interfaces.sh --nuke
+  ignore_errors: true
```

Does the rhel9 deployed OS not work with the clean-interfaces.sh script? Does it exist on the rhel9 Scale lab image/build?

---

It fails if the interfaces have already been cleaned.

---

Is that the intended behavior? Should we skip the clean interfaces for rhel9? We have to be very careful with the usage of `ignore_errors` when we can.

---

I'm not sure, it's different on RHEL8 vs RHEL9. It seems to fail silently on 8 and not on 9. I'll see if I can compare again.

---

Looks like the clean interfaces script has been rewritten to use nmcli on RHEL9.

---

Got it so in that case it returns an error code that ansible now picks up whereas with rhel8, it would fail but no error code was emitted and therefore not picked up by ansible?

---

Right, I guess nmcli considers it an error that it could not do what it was asked to do even if it's already been done, ie the interface is already down. Ifdown considers the interface already down to be success because we got the desired state.